py/persistentcode.h

```diff
@@ -30,24 +30,23 @@
 #include "py/reader.h"
 #include "py/emitglue.h"
 
-// The current version of .mpy files
+// The current version of .mpy files. A bytecode-only .mpy file can be loaded
+// as long as MPY_VERSION matches, but a native .mpy (i.e. one with an arch
+// set) must also match MPY_SUB_VERSION. This allows 3 additional updates to
+// the native ABI per bytecode revision.
 #define MPY_VERSION 6
+#define MPY_SUB_VERSION 1
```

Comment:
I like `MPY_SUB_VERSION`, but I did initially think it would be `MPY_VERSION_MINOR`